packages/starlight/__tests__/basics/format-path.test.ts

```diff
@@ -0,0 +1,105 @@
+		],
+	},
+	{
+		options: { format: 'file', trailingSlash: 'always' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/index.html' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration.html' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users.html' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components.html' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components.html' },
+		],
+	},
+	{
+		options: { format: 'file', trailingSlash: 'never' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/index.html' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration.html' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users.html' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components.html' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components.html' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'always' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration/' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users/' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components/' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components/' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'never' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components' },
+		],
+	},
+	{
+		options: { format: 'directory', trailingSlash: 'ignore' },
+		tests: [
+			// index page
+			{ path: '/', expected: '/' },
+			// with trailing slash
+			{ path: '/reference/configuration/', expected: '/reference/configuration/' },
+			// without trailing slash
+			{ path: '/api/v1/users', expected: '/api/v1/users' },
+			// with file extension
+			{ path: '/guides/components.html', expected: '/guides/components' },
+			// with file extension and trailing slash
+			{ path: '/guides/components.html/', expected: '/guides/components' },
+		],
+	},
+])(
+	'formatPath() with { format: $options.format, trailingSlash: $options.trailingSlash }',
+	({ options, tests }) => {
```

Comment:
Not sure I understand what this means. Could you explain a little more?

Reply:
Oh right. Yeah, this was cheating a bit just to avoid rewriting the tests too much. I think it’s fine though — it’s OK to do weird stuff in tests sometimes 😅